File: src/androidTest/java/org/tasks/injection/InjectingTestCase.java
package org.tasks.injection;

import org.junit.Before;

import static android.support.test.InstrumentationRegistry.getTargetContext;
import static org.tasks.TestUtilities.initializeMockito;

public abstract class InjectingTestCase {

    protected TestComponent component;

    @Before
    public void setUp() {
        initializeMockito(getTargetContext());

        component = DaggerTestComponent.builder()
                .testModule(new TestModule(getTargetContext()))
                .build();
        inject(component);
    }

    protected abstract void inject(TestComponent component);
}

File: app/src/androidTest/java/org/tasks/injection/InjectingTestCase.kt
package org.tasks.injection

import android.content.Context
import androidx.test.core.app.ApplicationProvider.getApplicationContext
import androidx.test.platform.app.InstrumentationRegistry
import dagger.hilt.android.testing.HiltAndroidRule
import org.junit.Before
import org.junit.Rule
import org.tasks.TestUtilities

abstract class InjectingTestCase {
    @get:Rule
    var hiltRule = HiltAndroidRule(this)

    @Before
    open fun setUp() {
        TestUtilities.newPreferences(context).clear()
        hiltRule.inject()
    }

    protected fun runOnMainSync(runnable: Runnable) =
            InstrumentationRegistry.getInstrumentation().runOnMainSync(runnable)

    protected val context: Context
        get() = getApplicationContext()
}